src/RE.cc

```diff
@@ -394,14 +408,22 @@ static RE_Matcher* matcher_merge(const RE_Matcher* re1, const RE_Matcher* re2, c
 	const char* text1 = re1->PatternText();
 	const char* text2 = re2->PatternText();
 
-	int n = strlen(text1) + strlen(text2) + strlen(merge_op) + 32 /* slop */;
+	size_t n = strlen(text1) + strlen(text2) + strlen(merge_op) + 32 /* slop */;
 
 	char* merge_text = new char[n];
 	snprintf(merge_text, n, "(%s)%s(%s)", text1, merge_op, text2);
 
 	RE_Matcher* merge = new RE_Matcher(merge_text);
 	delete[] merge_text;
 
+	// TODO: I'm not sure this is correct. If either of the joined matchers
+	// have the single-line flag set, then the conjuction will have it set.
+	// But that's not really right, since only the part that has the flag
+	// should be single-line. For example if you have /a.c/s | /d.f/, the
+	// dot should only match \n for the first part.
```

Comment:
I agree that this is a bit weird, and conceivably could be buggy.  I haven't yet looked at the rest of the changes, but I'd think a general approach would be for `/s` to result in translating (initially, not at compilation time) any `.`'s in the pattern to a new meta-character that means anything-including-newlines.  If that were done, then there's no need to explicitly propagate the SingleLine status.